packages/editor/src/lib/editor/Editor.ts

```diff
@@ -1011,16 +1041,69 @@ export class Editor extends EventEmitter<TLEventMap> {
 		return this
 	}
 
+	private _force = false
+
 	/**
-	 * Run a function in a batch.
+	 * Run a function in a transaction with optional options for context.
+	 * You can use the options to change the way that history is treated
+	 * or allow changes to locked shapes.
+	 *
+	 * @example
+	 * ```ts
+	 * // updating with
+	 * editor.run(() => {
+	 * 	editor.updateShape({ ...myShape, x: 100 })
+	 * }, { history: "ignore" })
+	 *
+	 * // forcing changes / deletions for locked shapes
+	 * editor.toggleLock([myShape])
+	 * editor.run(() => {
+	 * 	editor.updateShape({ ...myShape, x: 100 })
+	 * 	editor.deleteShape(myShape)
+	 * }, { ignoreShapeLock: true })
+	 * ```
+	 *
+	 * @param fn - The callback function to run.
+	 * @param opts - The options for the batch.
 	 *
 	 * @public
 	 */
-	batch(fn: () => void, opts?: TLHistoryBatchOptions): this {
-		this.history.batch(fn, opts)
+	run(
+		fn: () => void,
+		opts?: Partial<
+			TLHistoryBatchOptions & {
+				ignoreShapeLock: boolean
+			}
+		>
```

Comment:
In general, we should avoid `Partial` where its not needed because it produces much harder to read generated docs. Can we write this as `TLHistoryBatchOptions & {ignoreShapeLock?: boolean}` instead? Or even better:
```ts
interface TLBatchOptions extends TLHistoryBatchOptions {
    ignoreShapeLock?: boolean
}
```